### DISCUSSION TITLE
SmartWizard in Webpack

### DISCUSSION THREAD

**Discussion by garethfoxtrot** [techlab/jquery-smartwizard#126] [Q&A] [answered]
Hi, it's possible add SmartWizard library to Webpack? Actuallly I added lib to file app.js in webpack as:

require("smartwizard"); or require('smartwizard')( window, $ ); 
$('#smartwizard').SmartWizard();

but in browser is the following error: Uncaught TypeError: jquery__WEBPACK_IMPORTED_MODULE_2___default(...)(...).SmartWizard is not a function. 

It's possible fix this?

**Comment by techlab** [comment]
@garethfoxtrot I will check it, thanks for bringing it up.

**Reply by reidsneo** [reply]
any update?
I also occured a same issue

window.smartwizard = require('smartwizard');

$('#smartwizard').smartwizard();

jQuery.Deferred exception: $(...).smartwizard is not a function TypeError: $(...).smartwizard is not a function

tried using this project on laravel 8

**Comment by techlab** [comment, marked as answer]
Please follow the steps

### Install required modules
```bash
npm i jquery
npm i smartwizard

// Webpack
npm install webpack webpack-cli style-loader css-loader --save-dev
```

### Common JS/Webpack
```js
var $ = require( "jquery" );
require( "smartwizard/dist/css/smart_wizard_all.css");
const smartWizard = require("smartwizard");

$(function() {
    $('#smartwizard').smartWizard();
});
```
OR

### ES6/Babel
```js
import $ from "jquery";
import "smartwizard/dist/css/smart_wizard_all.css";
import smartWizard from 'smartwizard';

$(function() {
    $('#smartwizard').smartWizard();
});
```
